flipperdevices/flipperzero-firmware
# [FL-3289] Various Furi/FuriHal bug fixes and improvements

# What's new

- Furi: properly handle thread free before TCB scrapping
- FuriHal: add DWT comparator api to cortex
- Updater, RPC: refactor various thread shanenigans
- Reset debug and logging on production update
- Power: better low current gauge readings handling
- Code cleanup

# Verification 

- Ensure that no more crashes happens on reboot/update
- Ensure that RPC is not crashing on disconnect anymore
- Full release test is required

# Checklist (For Reviewer)
